SOSS/dms/soss_centroids.py

```diff
@@ -0,0 +1,377 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import os
+import warnings
+
+import numpy as np
+
+from astropy.io import fits
+
+from .soss_utils import zero_roll, robust_polyfit, get_image_dim
+
+from matplotlib import colors
+import matplotlib.pyplot as plt
+
+PATH = '/home/talens-irex/Dropbox/SOSS_Ref_Files'
+
+
+def _plot_centroid(image, xtrace, ytrace):
+    """Overplot the extracted trace positions on the image."""
+
+    nrows, ncols = image.shape
+
+    plt.figure(figsize=(ncols/128, nrows/128))
+
+    plt.imshow(image, origin='lower', cmap='inferno', norm=colors.LogNorm())
+    plt.plot(xtrace, ytrace, lw=2, ls='--', c='black', label='centroids')
+
+    plt.xlabel('Spectral Pixel', fontsize=14)
+    plt.ylabel('Spatial Pixel', fontsize=14)
+    plt.legend(fontsize=12)
+
+    plt.show()
+    plt.close()
+
+    return
+
+
+def center_of_mass(column, ypos, halfwidth):
```

Comment:
Have we had a discussion on with python version the dms is supporting?

If 3.7 or afterwards we should think about using the typing module
i.e.

```python
def center_of_mass(column: np.ndarray, ypos: float, halfwidth: int) --> float:
    """
    ...
    """
    ...
```
It's pretty similar to your `:type argument: type` but you just need to import List/Union/Tuple and use them correctly
```python
from typing import List, Tuple, Union

def test_func(x: List[str], y: Union[int, float, None] = None) -> Tuple[str, Union[int, float]]:
     return x[0], y

```


I can't believe how useful this is in finding bugs in pycharm since I switched to 3.7 (though again this depends completely on the DMS python version)

Reply:
I get that this is useful, but right now I just want to make headway on the reviews.

Reply:
Agreed, just noting it as it will make life easier unbreaking/debugging code